respect $XDG_CONFIG_HOME

@JulianAndrews makes a good point about supporting XDG_CONFIG_HOME on systems where it's set.

See https://github.com/exercism/exercism.io/issues/2212

Got to the point of running `glitch`, it fails with the following information:

```
2015/03/23 13:33:24 glitch: building
2015/03/23 13:33:24 glitch: build OK - vetting
2015/03/23 13:33:24 glitch: vet OK - testing
# github.com/exercism/cli/api
api/api_test.go:12:2: cannot find package "github.com/stretchr/testify/assert" in any of:
	/usr/lib/go/src/github.com/stretchr/testify/assert (from $GOROOT)
	/home/joneshf/programming/go/stuff/src/github.com/stretchr/testify/assert (from $GOPATH)
FAIL	github.com/exercism/cli/api [setup failed]
# github.com/exercism/cli/config
api/api_test.go:12:2: cannot find package "github.com/stretchr/testify/assert" in any of:
	/usr/lib/go/src/github.com/stretchr/testify/assert (from $GOROOT)
	/home/joneshf/programming/go/stuff/src/github.com/stretchr/testify/assert (from $GOPATH)
FAIL	github.com/exercism/cli/config [setup failed]
ok  	github.com/exercism/cli/cmd	0.002s
?   	github.com/exercism/cli/exercism	[no test files]
ok  	github.com/exercism/cli/user	0.002s
2015/03/23 13:33:24 glitch: failed
```

In any case, after installing  (just copying what the travis file does) I get more errors when I run `glitch` or `go test ./...`:

```
2015/03/23 13:36:43 glitch: building
2015/03/23 13:36:43 glitch: build OK - vetting
2015/03/23 13:36:43 glitch: vet OK - testing
ok  	github.com/exercism/cli/api	0.009s
ok  	github.com/exercism/cli/cmd	0.002s
--- FAIL: TestLoad (0.00s)
	config_test.go:21: link /home/joneshf/programming/go/stuff/src/github.com/exercism/cli/fixtures/config.json /tmp/877770574/config.json: invalid cross-device link
FAIL
FAIL	github.com/exercism/cli/config	0.003s
?   	github.com/exercism/cli/exercism	[no test files]
ok  	github.com/exercism/cli/user	0.002s
2015/03/23 13:36:44 glitch: failed
```

Same error either way.

--- a/config/config.go
+++ b/config/config.go
@@ -167,7 +167,14 @@ func (c *Config) resolvePath(argPath string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return strings.Replace(path, "~", h, 1), nil
+	path = strings.Replace(path, "~", h, 1)
+
+	fi, _ := os.Stat(path)
+	if fi != nil && fi.IsDir() {
+		path = filepath.Join(path, File)
+	}
+
+	return path, nil
 }
 
 func (c *Config) setDefaults() error {

--- a/config/config_test.go
+++ b/config/config_test.go
@@ -92,6 +92,25 @@ func TestLoad(t *testing.T) {
 	}
 }
 
+func TestReadDirectory(t *testing.T) {
+	// if the provided path is a directory, append the default filename
+	tmpDir, err := ioutil.TempDir("", "")
+	assert.NoError(t, err)
+
+	myConfig, err := New(tmpDir)
+	assert.NoError(t, err)
+
+	expected := filepath.Join(tmpDir, File)
+	actual := myConfig.File
+	assert.Equal(t, expected, actual)
+
+	// if it can't determine if the provided path is a directory, don't modify
+	// the path
+	myConfig, err = New("badpath")
+	assert.NoError(t, err)
+	assert.Equal(t, "badpath", myConfig.File)
+}
+
 func TestReadingWritingConfig(t *testing.T) {
 	tmpDir, err := ioutil.TempDir("", "")
 	filename := fmt.Sprintf("%s/%s", tmpDir, File)

--- a/exercism/main.go
+++ b/exercism/main.go
@@ -46,7 +46,7 @@ func main() {
 		cli.StringFlag{
 			Name:   "config, c",
 			Usage:  "path to config file",
-			EnvVar: "EXERCISM_CONFIG_FILE",
+			EnvVar: "XDG_CONFIG_HOME,EXERCISM_CONFIG_FILE",
 		},
 		cli.BoolFlag{
 			Name:  "verbose, v",

